### PR TITLE
feat(savedsearch): Add PUT for search_details endpoint

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Allows saved searches to be edited

### CODE REVIEW

File: src/sentry/api/endpoints/organization_search_details.py [hunk 1]
```diff
@@ -15,27 +17,54 @@
...
+        result = serializer.validated_data
+
+        if (
+            SavedSearch.objects
+            # Query duplication for pinned searches is fine, exlcuded these
+            .exclude(visibility=Visibility.OWNER_PINNED)
+            .filter(Q(is_global=True) | Q(organization=organization), query=result["query"])
+            .exists()
+        ):
+            return Response(
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_search_details.py:57]: Could use `ParseError` from rest_framework.exceptions.

File: tests/sentry/api/endpoints/test_organization_search_details.py [hunk 2]
```diff
@@ -85,3 +86,105 @@ def test_members_cannot_delete_shared_searches(self):
...
+            query="test",
+            visibility=Visibility.ORGANIZATION,
+        )
+        assert response.status_code == 200, response.content
+
+        updated_obj = SavedSearch.objects.get(id=search.id)
+        assert updated_obj.name == "foo"
+        assert updated_obj.query == "test"
+
+    def test_member_cannot_edit_org_search(self):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_search_details.py:127]: Should there be tests for the other organization roles?

File: tests/sentry/api/endpoints/test_organization_search_details.py [hunk 3]
```diff
@@ -85,3 +86,105 @@ def test_members_cannot_delete_shared_searches(self):
...
+        )
+        search = SavedSearch.objects.create(
+            organization=self.organization,
+            owner=self.user,
+            name="foo",
+            query="",
+            visibility=Visibility.OWNER,
+        )
+
+        # Ignores other users searches when updateing
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_search_details.py:169]:
```suggestion
        # Ignores other users searches when updating
```

File: src/sentry/api/endpoints/organization_search_details.py [hunk 4]
```diff
@@ -15,27 +17,54 @@
...
+        serializer = OrganizationSearchSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.validated_data
+
+        if (
+            SavedSearch.objects
+            # Query duplication for pinned searches is fine, exlcuded these
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_search_details.py:52]:
```suggestion
            # Query duplication for pinned searches is fine, excluded these
```

**Review by malwilley — APPROVED** [review]
Overall looks good to me, tested locally and everything works as intended.

This _does_ allow editing pinned searches which we don't allow in the UI but I can't think of a reason why we would need to prevent that.

**Comment by evanpurkhiser** [comment]
Ah, I might look into that actually

**Comment by evanpurkhiser** [comment]
> This does allow editing pinned searches

Nah it doesn't, this part in particular protects against that https://github.com/getsentry/sentry/pull/41010/files#diff-f6cf256101b7d54ec111badc5d92040674061f0f4384a66eadd789fbbc825021R26-R33

**Pull request merged**